test/mockcore_test.cc

```diff
@@ -73,33 +99,26 @@ class MockCoreFixture : public ::testing::Test {
         while (mGameManager.HasEventToProcess()) {}
     }
 
-    void RegisterAndGameStart(int roomId, const std::vector<unsigned int> &userIds) {
+    void RegisterAndGameStart(int roomId, 
+        const std::vector<unsigned int> &userIds) 
+    {
         EXPECT_CALL(*mMockStream, Write(_, _)).Times(AtLeast(1));
         mNetworkInterface->SendMsg(
             *MsgBuilder::CreateRegisterArgs(0, "uno", "UNO", 4, 2));
         ProcessMsgFromCore(*MsgBuilder::CreateRegisterRet(0, ErrorNumber::OK));
 
         /// TODO: check flippedCard and firstPlayerId with action
         EXPECT_CALL(*mMockStream, Write(
-            AllOf(
-                NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, userIds[0]),
-                GameStartMatcher()
-            ), _)
-        ).Times(1);
+            AllOf(NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, userIds[0]),
+                GameStartMatcher()), _)).Times(1);
 
         EXPECT_CALL(*mMockStream, Write(
-            AllOf(
-                NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, userIds[1]),
-                GameStartMatcher()
-            ), _)
-        ).Times(1);
+            AllOf(NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, userIds[1]),
+                GameStartMatcher()), _)).Times(1);
 
         EXPECT_CALL(*mMockStream, Write(
-            AllOf(
-                NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, userIds[2]),
-                GameStartMatcher()
-            ), _)
-        ).Times(1);
+            AllOf(NotifyMsgArgsMatcher(ErrorNumber::OK, roomId, userIds[2]),
```

Comment:
not sure if you send different msg to these users.  If you just want to do a broadcast, use 0 as the notifyMsg target user. Core will broadcast that msg to each user in the room.

Reply:
yep, clients receive different `GameStart` msgs (which is oneof `NotifyMsg`), because initial handcards info are included in that msg, which are different for each client

Reply:
btw, as for other `NotifyMsg` type, by design the principle is to broadcast the `UserOperation` to all clients except the one who did the operation and if the `UserOperation` needs to respond, send back a `*Rsp`. to be precise,
+ client 1 sends a `Draw`, provider needs to send back to client 1 a `DrawRsp` and to client 0, 2 a `Draw`
+ client 2 sends a `Skip`, provider forwards that `Skip` to client 0, 1 
+ client 0 sends a `Play`, provider forwards that `Play` to client 1, 2